dashboard/models/user.py

```diff
@@ -24,59 +24,75 @@ def api_url(self):
         if self.app_config is not None:
             return self.app_config.MOZILLIANS_API_URL
 
-    def _has_avatar(self, email):
-        if self.api_url is not None:
-            try:
-                mozillians_response = requests.get(self.api_url, headers=self.headers,
-                                                   params=self.params, timeout=5)
-                if mozillians_response.status_code is not 200:
-                    return None
-                response = mozillians_response.json()
-                return response
-            except (requests.exceptions.Timeout, requests.exceptions.ConnectionError):
-                return None
-        else:
+    def _get_user_info(self, email):
+        response = self._get_user(email)
+        if not response:
             return None
 
-    def _is_only_one_avatar(self, response):
-        # Check if only single resource gets returned and it's valid
-        avatars = response.get('results', -1)
+        try:
+            results = response.get('results', -1)
+            user_url = results[0].get('_url')
+            user_info = requests.get(user_url, headers=self.headers, timeout=5)
```

Comment:
I would suggest to have in the try block only the line that might throw an exception.